web-server/assets/src/script/skills.js

```diff
@@ -76,7 +76,24 @@ const filterSkillsList = filter => {
   })
 }
 
+const hideEmptyFilters = (filterElements) => {
+  filterElements.each((index, filter) => {
+    const filters = ($(filter).text() || '').trim().toLowerCase().split(/\s+/)
+    let hasSkills = false
+    $('.skills-list .skills-list-list > li').each((i, skill) => {
```

Comment:
store the value of `$('.skills-list .skills-list-list > li')` before the `each` loop. its expensive to query the dom like this and the results shouldn't change.

Reply:
something like `const skills = $('.skills-list .skills-list-list > li').map(n => $(n).text() || '')` would get you an array of the text of each skill.